sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIOTest.java

```diff
@@ -594,18 +731,63 @@ public void testReadingWithFilterAndSubSplits() throws Exception {
     makeTableData(table, numRows);
     service.setupSampleRowKeys(table, numSamples, bytesPerRow);
 
+    List<ByteKeyRange> keyRanges = new ArrayList<>();
+    keyRanges.add(ByteKeyRange.ALL_KEYS);
     // Generate source and split it.
     RowFilter filter =
         RowFilter.newBuilder().setRowKeyRegexFilter(ByteString.copyFromUtf8(".*17.*")).build();
     BigtableSource source =
-        new BigtableSource(serviceFactory, table, filter, ByteKeyRange.ALL_KEYS, null /*size*/);
+        new BigtableSource(serviceFactory, table, filter, keyRanges, null /*size*/);
     List<BigtableSource> splits = source.split(numRows * bytesPerRow / numSplits, null);
 
     // Test num splits and split equality.
     assertThat(splits, hasSize(numSplits));
     assertSourcesEqualReferenceSource(source, splits, null /* options */);
   }
 
+  /** Tests reading all rows from a sub-split table with a filter and several key ranges. */
```

Comment:
Here too can you clarify what additional test coverage this provides?

Reply:
I don't think this one does anything special. Removing.